diff --git a/chrome/browser/ui/BUILD.gn b/chrome/browser/ui/BUILD.gn
index 5ef79ef..fcaff61 100644
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -104,6 +104,8 @@ static_library("ui") {
     "webui/data_sharing_internals/data_sharing_internals_ui.h",
     "webui/device_log/device_log_ui.cc",
     "webui/device_log/device_log_ui.h",
+    "webui/distraction_blocked/distraction_blocked_ui.cc",
+    "webui/distraction_blocked/distraction_blocked_ui.h",
     "webui/download_internals/download_internals_ui.cc",
     "webui/download_internals/download_internals_ui.h",
     "webui/download_internals/download_internals_ui_message_handler.cc",
